x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -278,6 +279,19 @@ export class TaskRunner<
 
     const ruleLabel = `${this.ruleType.id}:${ruleId}: '${name}'`;
 
+    if (this.context.alertsService && ruleType.alerts) {
+      // Wait for alerts-as-data resources to be installed
+      // Since this occurs at the beginning of rule execution, we can be
+      // assured that all resources will be ready for reading/writing when
+      // the rule type executors are called
+
+      // TODO - add retry if any initialization steps have failed
+      await this.context.alertsService.getContextInitializationPromise(
```

Comment:
**question:** From what I can tell, the promise resolves to the following two properties: `{ result, error }`. If ever result (initialized) is false, should the rules keep running?

Reply:
Currently, if an error is returned, this means the AAD resources aren't initialized and the rule registry will throw an error when the lifecycle/persistence executors try to get a writer. The alerting framework logic will still work as before, so we're keeping the existing functionality, where the alert documents might not get written but any notifications would still get sent. When we move to Phase 2 and start writing out alert documents in the framework, we should be more reactive to errors during initialization.

Reply:
That makes sense to me, thank you! Resolving..